src/libcore/slice/mod.rs

```diff
@@ -1215,6 +1215,22 @@ macro_rules! iterator {
                 }
                 accum
             }
+
+            #[inline]
+            fn position<P>(&mut self, mut predicate: P) -> Option<usize> where
+                Self: Sized,
+                P: FnMut(Self::Item) -> bool,
+            {
+                // The addition might panic on overflow
+                self.try_fold(0, move |i, x| {
+                    if predicate(x) { LoopState::Break(i) }
```

Comment:
failed to resolve. Use of undeclared type or module `LoopState`

Reply:
I'm changing it to use Result<,>'s instead.